js/modules/k6/experimental/fs/module_test.go

```diff
@@ -0,0 +1,695 @@
+			try {
+				const file = await fs.open(%q)
+				throw 'unexpected promise resolution with result: ' + res
+			} catch (err) {
+				if (err.name !== 'InvalidResourceError') {
+					throw 'unexpected error: ' + err
+				}
+			}
+		`, testDirPath)))
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("opening non existing file should fail", func(t *testing.T) {
+		t.Parallel()
+
+		runtime, err := newConfiguredRuntime(t)
+		require.NoError(t, err)
+
+		_, err = runtime.RunOnEventLoop(wrapInAsyncLambda(`
+			try {
+				const file = await fs.open('doesnotexist.txt')
+				throw 'unexpected promise resolution with result: ' + res
+			} catch (err) {
+				if (err.name !== 'NotFoundError') {
+					throw 'unexpected error: ' + err
+				}
+			}
+		`))
+
+		assert.NoError(t, err)
+	})
+}
+
+func TestFile(t *testing.T) {
+	t.Parallel()
+
+	t.Run("stat method should succeed", func(t *testing.T) {
+		t.Parallel()
+
+		runtime, err := newConfiguredRuntime(t)
+		require.NoError(t, err)
+
+		testFilePath := fsext.FilePathSeparator + testFileName
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, testFilePath, []byte("Bonjour, le monde"), 0o644)
+		})
+		runtime.VU.InitEnvField.FileSystems["file"] = fs
+
+		_, err = runtime.RunOnEventLoop(wrapInAsyncLambda(fmt.Sprintf(`
+			const file = await fs.open(%q)
+			const info = await file.stat()
+
+			if (info.name !== 'bonjour.txt') {
+				throw 'unexpected file name ' + info.name + '; expected \'bonjour.txt\'';
+			}
+
+			if (info.size !== 17) {
+				throw 'unexpected file size ' + info.size + '; expected 17';
+			}
+		`, testFilePath)))
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("read in multiple iterations", func(t *testing.T) {
+		t.Parallel()
+
+		runtime, err := newConfiguredRuntime(t)
+		require.NoError(t, err)
+
+		testFilePath := fsext.FilePathSeparator + testFileName
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, testFilePath, []byte("01234"), 0o644)
+		})
+		runtime.VU.InitEnvField.FileSystems["file"] = fs
+
+		_, err = runtime.RunOnEventLoop(wrapInAsyncLambda(fmt.Sprintf(`
+			const file = await fs.open(%q);
+
+			let fileContent = new Uint8Array(5);
+
+			let bytesRead;
+			let buffer = new Uint8Array(3);
+
+			bytesRead = await file.read(buffer)
+			if (bytesRead !== 3) {
+				throw 'expected read to return 3, got ' + bytesRead + ' instead';
+			}
+
+			// We expect the buffer to be filled with the three first
+			// bytes of the file, namely '012'.
+			if (buffer[0] !== 48 || buffer[1] !== 49 || buffer[2] !== 50) {
+				throw 'expected buffer to be [48, 49, 50], got ' + buffer + ' instead';
+			}
+
+			fileContent.set(buffer, 0);
+
+			bytesRead = await file.read(buffer)
+			if (bytesRead !== 2) {
+				throw 'expected read to return 2, got ' + bytesRead + ' instead';
+			}
+
+			// We expect the buffer to hold the two last bytes of the
+			// file, namely '34', and as we read only two bytes, its last
```

Comment:
nit: "namely '34'" and then below there is no '34'

Not certain if this used to say '34' or not. Maybe we can leave only the part about the last one not being changed :shrug:

Reply:
Fair point 👍